macros/src/types/enum.rs

```diff
@@ -115,18 +115,20 @@ fn format_variant(
                     ..
                 } = FieldAttr::from_attrs(&unnamed.unnamed[0].attrs)?;
 
-                let ty = match (type_override, type_as) {
-                    (Some(_), Some(_)) => syn_err!("`type` is not compatible with `as`"),
-                    (Some(type_override), None) => quote! { #type_override },
-                    (None, Some(type_as)) => {
-                        format_type(&syn::parse_str::<Type>(&type_as)?, dependencies, generics)
-                    }
-                    (None, None) => format_type(&unnamed.unnamed[0].ty, dependencies, generics),
-                };
+
 
                 if skip {
                     quote!(format!("{{ \"{}\": \"{}\" }}", #tag, #name))
                 } else {
+                    let ty = match (type_override, type_as) {
+                        (Some(_), Some(_)) => syn_err!("`type` is not compatible with `as`"),
+                        (Some(type_override), None) => quote! { #type_override },
+                        (None, Some(type_as)) => {
+                            format_type(&syn::parse_str::<Type>(&type_as)?, dependencies, generics)
+                        }
+                        (None, None) => format_type(&unnamed.unnamed[0].ty, dependencies, generics),
+                    };
+
                     quote!(format!("{{ \"{}\": \"{}\", \"{}\": {} }}", #tag, #name, #content, #ty))
                 }
```

Comment:
@timephy This here was the issue: `format_type` was called outside of the `if skip { .. }`, adding the content to the dependencies even if the field was to be skipped.

Reply:
Hey :)
Great investigation! Thank you for the quick fix - highly appreciated!